core/retries-api/src/main/java/software/amazon/awssdk/retriesapi/TokenAcquisitionFailedException.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.retriesapi;
+
+import software.amazon.awssdk.annotations.SdkPublicApi;
+
+/**
+ * Exception thrown by {@link RetryStrategy} when a new token cannot be acquired.
+ */
+@SdkPublicApi
+public final class TokenAcquisitionFailedException extends RuntimeException {
```

Comment:
I wonder if it should extend SdkClientException to make it easier for customers to handle

Reply:
I guess we could depend on `core`, but not sure if we want to complect those. This package for now only depends on `annotations` and `utils` and I feel that adding `sdk-core` will make it a lot less easier to reuse outside the SDK.